packages/expo-cli/src/commands/client/index.js

```diff
@@ -29,6 +29,18 @@ export default program => {
     .asyncAction(async options => {
       const authData = await appleApi.authenticate(options);
       const user = await User.getCurrentUserAsync();
+
+      // check if any builds are in flight
+      const { isAllowed, errorMessage } = await isAllowedToBuild({
+        user,
+        appleTeamId: authData.team.id,
+      });
+
+      if (!isAllowed) {
+        log.error(`New Expo client build request disallowed. Reason: ${errorMessage}`);
```

Comment:
Maybe it would be better to throw an error here? I think `asyncAction` handles that case. 
Why should it be better? In case you don't throw an error (and just return from this function) expo-cli will exit with `0` code which states that there were no errors with the program execution. Throwing an error will set the exit code properly.